### DISCUSSION TITLE
Recommended Approach to Wait for a PDF to Load?

### DISCUSSION THREAD

**Discussion by Mike-E-angelo** [hardkoded/puppeteer-sharp#2712] [Q&A]
Hello Community,

This is tangentially-related to [this discussion](https://github.com/hardkoded/puppeteer-sharp/discussions/2696).  I am encountering an issue with taking screenshots of PDF files.  Currently, I have to use a `Task.Delay(5000)` to wait for the PDF to fully load before I can take a screenshot.  Any time before this the PDF is not fully loaded and results in a blurry circle instead of a screenshot of the content.

As this time seems to be dependent on the speed of the machine I am worried that this wait will be different upon deployment and still result in inaccurate screenshots. Is there an elegant way to wait for the PDF to fully load before taking the screenshot?

Thank you for any assistance you can provide.

**Comment by CaCTuCaTu4ECKuu** [comment]
Are you talking about like opening .pdf using browser and want to rasterize it from there? Are you sure you dont want to use small specialized library for that?
As for opening PDFs in browser I can imagine, there is pretty large files that even last gen PCs need a moment to process so you'll definitely will have a problem with this approach so my guess there should be some king of event on load complete. Maybe opening file is treated like loading a file so you can track it like network connection, did you check it?

**Reply by Mike-E-angelo** [reply]
Pardon the confusion, it's asking Puppeteer to load the PDF (by visiting it/loading it as the URL) and then waiting for it to be fully/completly loaded before calling the `ScreenshotAsync` method.  If I load the PDF address/location and immediately call `ScreenshotAsync` then it's just a blurry circle.  If I put in the `Task.Delay` then the image is as expected.  I'd rather not have the `Task.Delay` but be notified when the page is completely loaded.
